jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/parser/HeaderBlockParser.java

```diff
@@ -135,4 +139,68 @@ public MetaData parse(ByteBuffer buffer, int blockLength)
             }
         }
     }
+
+    public static class SessionFailureMetaData extends MetaData
+    {
+        private SessionFailureMetaData()
+        {
+            super(HttpVersion.HTTP_2, HttpFields.EMPTY);
+        }
+    }
+
+    public interface StreamFailureMetaData
```

Comment:
javadoc

Reply:
Why is both a session and stream failure needed?   Can't a session failure be detected by the fact that it has a failed stream?

Or perhaps it could just be a FailureMetaData with reason in the getFailure?